Sage-Bionetworks/BridgePF
# Configurable Auth Migration DAO which reads from MySQL, falls back to Stormpath, and writes to both

This is configurable and has three settings:
* Stormpath only
* Create in Stormpath only, but read and update both Stormpath and MySQL - This is used to test the backwards compatible case where MySQL is turned on, but the accounts don't exist in MySQL yet.
* Full MySQL - Falls back to Stormpath and writes to both.

Testing done:
* updated unit tests
* ran integ tests for all 3 cases
* manual tests for all 3 cases

Some manual tests aren't currently testable in integ tests, or we wanted to take special care to verify account data that's not trackable through integ tests:
* SignUp (receive verification), Resend Verification, Verify, SignIn
* Request Reset Password, Reset Password
* EmailSignIn w/ change password
* consents, attributes, roles, participant options such as external ID (to both sources)
* Account disabled

Again, sorry for the massive changelist. There wasn't any way to make it smaller.
